### PR TITLE
Add fallback for filePath to errorToDiagnostic

### PR DESCRIPTION
**Pull request by DeMoorJasper**
# ↪️ Pull Request

Fallback to the asset/bundle filepath when converting a caught exception to a diagnostic.

Related to https://github.com/parcel-bundler/parcel/issues/5773#issuecomment-775640779

### CODE REVIEW

**Comment by parcel-benchmark** [comment]
## Benchmark Results
<details><summary>Kitchen Sink ✅</summary><p>

#### Timings

| Description | Time | Difference |
| --- | --- | --- |
| Cold | 2.47s | -37.00ms |
| Cached | 519.00ms | -4.00ms |

#### Cold Bundles

| Bundle | Size | Difference | Time | Difference |
| --- | --- | --- | --- | --- |
| dist/modern/parcel.d5807e82.webp | 102.94kb | +0.00b | 566.00ms | -37.00ms 🚀 |
| dist/modern/index.9e1f1458.js | 1.11kb | +0.00b | 566.00ms | -41.00ms 🚀 |
| dist/modern/index.html | 701.00b | +0.00b | 566.00ms | -39.00ms 🚀 |
| dist/legacy/index.44edca46.css | 77.00b | +0.00b | 922.00ms | +74.00ms ⚠️ |
| dist/modern/index.0bc656bc.css | 77.00b | +0.00b | 920.00ms | -61.00ms 🚀 |

#### Cached Bundles

| Bundle | Size | Difference | Time | Difference |
| --- | --- | --- | --- | --- |
| dist/modern/parcel.d5807e82.webp | 102.94kb | +0.00b | 49.00ms | -20.00ms 🚀 |
| dist/modern/index.9e1f1458.js | 1.11kb | +0.00b | 49.00ms | -20.00ms 🚀 |
| dist/legacy/index.html | 701.00b | +0.00b | 50.00ms | +3.00ms ⚠️ |
| dist/modern/index.html | 701.00b | +0.00b | 49.00ms | -20.00ms 🚀 |
| dist/legacy/index.44edca46.css | 77.00b | +0.00b | 50.00ms | +3.00ms ⚠️ |
| dist/modern/index.0bc656bc.css | 77.00b | +0.00b | 49.00ms | -20.00ms 🚀 |

</p></details>

<details><summary>React HackerNews ✅</summary><p>

#### Timings

| Description | Time | Difference |
| --- | --- | --- |
| Cold | 16.97s | +16.97s ⚠️ |
| Cached | 7.38s | +7.38s ⚠️ |

#### Cold Bundles

*No bundles found, this is probably a failed build...*
#### Cached Bundles

*No bundles found, this is probably a failed build...*
</p></details>

<details><summary>AtlasKit Editor ✅</summary><p>

#### Timings

| Description | Time | Difference |
| --- | --- | --- |
| Cold | 1.95m | +14.70s ⚠️ |
| Cached | 2.55s | -3.00ms |

#### Cold Bundles

| Bundle | Size | Difference | Time | Difference |
| --- | --- | --- | --- | --- |
| dist/index.d51fa13e.js | 2.46mb | +0.00b | 1.19m | +15.71s ⚠️ |
| dist/editorView.2ca486fc.js | 587.53kb | +0.00b | 27.06s | -22.00s 🚀 |
| dist/popup.23224432.js | 169.24kb | +0.00b | 15.38s | -819.00ms 🚀 |
| dist/EmojiPickerComponent.90b741d3.js | 139.01kb | +0.00b | 9.11s | -752.00ms 🚀 |
| dist/Toolbar.57f2e74b.js | 99.69kb | +0.00b | 12.39s | -3.81s 🚀 |
| dist/media-viewer.d1cb7a32.js | 72.87kb | +0.00b | 9.18s | -1.01s 🚀 |
| dist/card.ecae978d.js | 54.02kb | +0.00b | 7.70s | -410.00ms 🚀 |
| dist/card.86f6e816.js | 51.77kb | +0.00b | 15.20s | +4.34s ⚠️ |
| dist/Modal.a2c499cb.js | 39.18kb | +0.00b | 3.10s | -396.00ms 🚀 |
| dist/component.fa4e42d1.js | 30.82kb | +0.00b | 3.35s | +180.00ms ⚠️ |
| dist/esm.6bc5d42c.js | 27.67kb | +0.00b | 7.53s | -915.00ms 🚀 |
| dist/component.63b8a376.js | 22.54kb | +0.00b | 3.08s | -407.00ms 🚀 |
| dist/smartMediaEditor.70ca3661.js | 16.85kb | +0.00b | 12.38s | -3.79s 🚀 |
| dist/js.4b5d27df.js | 16.55kb | +0.00b | 3.08s | -407.00ms 🚀 |
| dist/workerHasher.09e24457.js | 11.90kb | +0.00b | 5.36s | -332.00ms 🚀 |
| dist/card.1a9729d8.js | 5.71kb | +0.00b | 9.19s | +625.00ms ⚠️ |
| dist/media-viewer.941270ab.js | 4.01kb | +0.00b | 9.16s | -945.00ms 🚀 |
| dist/EmojiPickerComponent.ca69a3c9.js | 3.62kb | +0.00b | 5.41s | -1.01s 🚀 |
| dist/png-chunks-extract.e6351d92.js | 3.55kb | +0.00b | 3.08s | -407.00ms 🚀 |
| dist/index.ece3a13e.css | 3.46kb | +0.00b | 13.03s | -42.50s 🚀 |
| dist/Modal.00beb515.js | 3.14kb | +0.00b | 872.00ms | -222.00ms 🚀 |
| dist/16.cf92dee6.js | 2.49kb | +0.00b | 872.00ms | -222.00ms 🚀 |
| dist/ResourcedEmojiComponent.0df891ba.js | 2.15kb | +0.00b | 6.17s | -1.54s 🚀 |
| dist/card.b6509036.js | 2.15kb | +0.00b | 9.40s | +593.00ms ⚠️ |
| dist/date.f35453a2.js | 1.96kb | +0.00b | 3.61s | -306.00ms 🚀 |
| dist/images.861d8ede.js | 1.90kb | +0.00b | 3.62s | -609.00ms 🚀 |
| dist/feedback.ddd45d32.js | 1.86kb | +0.00b | 4.97s | -2.82s 🚀 |
| dist/16.4dbc75f9.js | 1.86kb | +0.00b | 3.41s | -624.00ms 🚀 |
| dist/browser.ba398ea7.js | 1.82kb | +0.00b | 9.57s | -1.19s 🚀 |
| dist/16.934d2c3d.js | 1.79kb | +0.00b | 3.24s | -430.00ms 🚀 |
| dist/workerHasher.afb52a72.js | 1.75kb | +0.00b | 7.52s | +1.85s ⚠️ |
| dist/workerHasher.42458950.js | 1.75kb | +0.00b | 15.28s | +4.34s ⚠️ |
| dist/list-number.95becf64.js | 1.68kb | +0.00b | 3.73s | -298.00ms 🚀 |
| dist/status.5568e78d.js | 1.68kb | +0.00b | 3.85s | -628.00ms 🚀 |
| dist/code.94d6ed50.js | 1.61kb | +0.00b | 3.53s | -385.00ms 🚀 |
| dist/link.b4c4afb6.js | 1.53kb | +0.00b | 3.73s | -298.00ms 🚀 |
| dist/heading6.b56f3cf2.js | 1.53kb | +0.00b | 3.99s | -1.34s 🚀 |
| dist/heading3.0c0fe1f4.js | 1.51kb | +0.00b | 3.97s | -287.00ms 🚀 |
| dist/16.2a10e9af.js | 1.51kb | +0.00b | 3.40s | -234.00ms 🚀 |
| dist/16.d564a7d9.js | 1.46kb | +0.00b | 3.24s | -350.00ms 🚀 |
| dist/16.01c01dd6.js | 1.46kb | +0.00b | 3.26s | -375.00ms 🚀 |
| dist/emoji.4f12a585.js | 1.45kb | +0.00b | 3.61s | -306.00ms 🚀 |
| dist/16.81d1aa0c.js | 1.45kb | +0.00b | 872.00ms | -222.00ms 🚀 |
| dist/16.897e8807.js | 1.44kb | +0.00b | 3.53s | -253.00ms 🚀 |
| dist/16.c2dce044.js | 1.40kb | +0.00b | 3.53s | -254.00ms 🚀 |
| dist/heading5.bde11017.js | 1.40kb | +0.00b | 3.97s | -286.00ms 🚀 |
| dist/16.ac96fdd0.js | 1.36kb | +0.00b | 3.68s | -351.00ms 🚀 |
| dist/16.620d8c15.js | 1.34kb | +0.00b | 872.00ms | -222.00ms 🚀 |
| dist/heading2.1511a751.js | 1.33kb | +0.00b | 3.97s | -287.00ms 🚀 |
| dist/16.990be6f2.js | 1.31kb | +0.00b | 3.53s | -253.00ms 🚀 |
| dist/mention.e2501ad0.js | 1.29kb | +0.00b | 3.73s | -292.00ms 🚀 |
| dist/heading4.1abfeb0a.js | 1.29kb | +0.00b | 3.97s | -287.00ms 🚀 |
| dist/Modal.b44f65e1.js | 1.28kb | +0.00b | 3.08s | -408.00ms 🚀 |
| dist/16.30ebc170.js | 1.27kb | +0.00b | 872.00ms | -222.00ms 🚀 |
| dist/16.3ba7b0ec.js | 1.27kb | +0.00b | 3.40s | -234.00ms 🚀 |
| dist/16.2ec03e4f.js | 1.26kb | +0.00b | 3.24s | -595.00ms 🚀 |
| dist/16.b9cdc131.js | 1.26kb | +0.00b | 3.40s | -234.00ms 🚀 |
| dist/16.98030c60.js | 1.26kb | +0.00b | 3.40s | -234.00ms 🚀 |
| dist/16.c20ef86c.js | 1.26kb | +0.00b | 3.24s | -349.00ms 🚀 |
| dist/divider.9a733dfe.js | 1.25kb | +0.00b | 3.61s | -306.00ms 🚀 |
| dist/quote.be70ca39.js | 1.25kb | +0.00b | 3.85s | -628.00ms 🚀 |
| dist/action.88cd1e67.js | 1.23kb | +0.00b | 3.53s | -524.00ms 🚀 |
| dist/decision.0740ec10.js | 1.21kb | +0.00b | 3.61s | -306.00ms 🚀 |
| dist/panel-warning.9b047bba.js | 1.21kb | +0.00b | 3.85s | -312.00ms 🚀 |
| dist/list.e4113e60.js | 1.18kb | +0.00b | 3.73s | -292.00ms 🚀 |
| dist/panel-error.0c73ee4f.js | 1.11kb | +0.00b | 3.74s | -426.00ms 🚀 |
| dist/panel.53ea0317.js | 1.10kb | +0.00b | 3.85s | -312.00ms 🚀 |
| dist/table.23e2141c.js | 1.09kb | +0.00b | 3.86s | -642.00ms 🚀 |
| dist/media-card-analytics-error-boundary.64c19024.js | 999.00b | +0.00b | 7.50s | -715.00ms 🚀 |
| dist/media-card-analytics-error-boundary.60a98f4f.js | 999.00b | +0.00b | 9.84s | -1.03s 🚀 |
| dist/simpleHasher.931d48e8.js | 755.00b | +0.00b | 5.36s | -340.00ms 🚀 |

#### Cached Bundles

| Bundle | Size | Difference | Time | Difference |
| --- | --- | --- | --- | --- |
| dist/Modal.00beb515.js | 3.14kb | +0.00b | 488.00ms | -31.00ms 🚀 |
| dist/16.cf92dee6.js | 2.49kb | +0.00b | 488.00ms | -30.00ms 🚀 |
| dist/16.81d1aa0c.js | 1.45kb | +0.00b | 488.00ms | -31.00ms 🚀 |
| dist/16.620d8c15.js | 1.34kb | +0.00b | 488.00ms | -30.00ms 🚀 |
| dist/16.30ebc170.js | 1.27kb | +0.00b | 488.00ms | -30.00ms 🚀 |
| dist/index.html | 119.00b | +0.00b | 471.00ms | +68.00ms ⚠️ |

</p></details>

<details><summary>Three.js ✅</summary><p>

#### Timings

| Description | Time | Difference |
| --- | --- | --- |
| Cold | 12.06s | -23.00ms |
| Cached | 515.00ms | +5.00ms |

#### Cold Bundles

*No bundle changes detected.*

#### Cached Bundles

| Bundle | Size | Difference | Time | Difference |
| --- | --- | --- | --- | --- |
| dist/Three.js | 580.59kb | +0.00b | 81.00ms | -6.00ms 🚀 |

</p></details>

**[Click here to view a detailed benchmark overview.](https://benchmark.parceljs.org/benchmark/290073039049064968)**

File: packages/core/diagnostic/src/diagnostic.js [hunk 1]
```diff
@@ -119,14 +119,18 @@ export function anyToDiagnostic(input: Diagnostifiable): Array<Diagnostic> {
 /** Normalize the given error into a diagnostic. */
 export function errorToDiagnostic(
   error: ThrowableDiagnostic | PrintableError | string,
-  realOrigin?: string,
+  defaultValues: {
+    origin?: ?string,
+    filePath?: ?string,
+    ...
```
Comment (mischnic) [inline packages/core/diagnostic/src/diagnostic.js:125]: Could this be an exact object type instead?

Comment (DeMoorJasper) [reply packages/core/diagnostic/src/diagnostic.js]: Sure

File: packages/core/diagnostic/src/diagnostic.js [hunk 2]
```diff
@@ -119,14 +119,17 @@ export function anyToDiagnostic(input: Diagnostifiable): Array<Diagnostic> {
 /** Normalize the given error into a diagnostic. */
 export function errorToDiagnostic(
   error: ThrowableDiagnostic | PrintableError | string,
-  realOrigin?: string,
+  defaultValues: {|
+    origin?: ?string,
+    filePath?: ?string,
+  |} = {},
```
Comment (mischnic) [inline packages/core/diagnostic/src/diagnostic.js:125]:
```suggestion
  |} = {...null},
```

Thanks flow...

**Review by mischnic — APPROVED** [review]

**Review by devongovett — APPROVED** [review]

**Pull request merged**